lib/web/join_tokens.go

```diff
@@ -62,7 +62,22 @@ func (h *Handler) createScriptJoinTokenHandle(w http.ResponseWriter, r *http.Req
 		return nil, trace.Wrap(err)
 	}
 
-	return createScriptJoinToken(r.Context(), clt)
+	roles := types.SystemRoles{
+		types.RoleNode,
+		types.RoleApp,
+	}
+
+	return createJoinToken(r.Context(), clt, roles)
+}
+
+func (h *Handler) createDatabaseJoinTokenHandle(w http.ResponseWriter, r *http.Request, params httprouter.Params, ctx *SessionContext) (interface{}, error) {
```

Comment:
Please create a test for the new endpoint. Let's exercise it via a client instead of calling private methods (which I see some of the existing tests too).

Reply:
Sure, I added it in this commit: https://github.com/gravitational/teleport/pull/10914/commits/103a89403f43bf02c6d99553bf7940e99a3de834

I'm calling the endpoint and checking that a valid token with role `types.RoleDatabase` is created.

I had to add permission to create tokens to this [method that creates an admin](https://github.com/gravitational/teleport/blob/103a89403f43bf02c6d99553bf7940e99a3de834/lib/services/role.go#L124) user for testing. 
Since this is only used on tests and all tests are passing, I think this is ok.

Reply:
> Note that I had to add permission to create tokens to this [method that creates an admin](https://github.com/gravitational/teleport/blob/103a89403f43bf02c6d99553bf7940e99a3de834/lib/services/role.go#L124) user for testing.

It seems alright, but if you need specific users/roles in the future, consider creating them in the test body.